content/en/continuous_integration/pipelines/awscodepipeline.md

```diff
@@ -0,0 +1,126 @@
+1. In the AWS Console, go to **EventBridge > Rules** and click on **Create Rule**.
+2. Choose a name for the rule (for example, **datadog-ci-visibility-integration**) and optionally add a description.
+3. Leave the event bus as **default**, and, under **Rule Type**, select **Rule with an event pattern**. Click on **Next**.
+4. Under **Event Source**, select **AWS events or EventBridge partner events**.
+5. Under **Creation Method**, select **Custom pattern (JSON editor)**. Then, under **Event Pattern**, input the following JSON:
+   ```
+   {
+     "source": ["aws.codepipeline"],
+     "detail-type": ["CodePipeline Pipeline Execution State Change", "CodePipeline Action Execution State Change", "CodePipeline Stage Execution State Change"]
+   }
+   ```
+   The JSON above sets up the integration for all of your pipelines. To restrict the set of pipelines,
+   follow the [Only monitor specific pipelines][6] section below.
+6. Click on **Next**.
+7. Under **Target Types**, select **EventBridge API destination**. Then, choose **Use an existing API Destination**
+and select the API destination that you have created in the previous step. Alternatively, you can also create the API destination
+at this moment by following the steps outlined in the [Create the API Destination][7] section.
+8. Under **Headers Parameters**, Click on **Add header parameter**. Input **DD-CI-PROVIDER-AWSCODEPIPELINE** as the key and **true** as the value.
+9. Choose **Create a new role for this specific resource** (or use an existing one if you prefer).
+10. Review that the information is correct and create the rule.
+
+Once the rule is created, the integration is complete, and you can monitor your pipelines in Datadog.
+
+### Only monitor specific pipelines
+
+You can optionally restrict the pipelines that are monitored by the CI Visibility product.
+To do this, add the `detail.pipeline` filter in the rule event pattern defined when creating the EventBridge Rule. For example:
+
+```
+ {
+   "source": ["aws.codepipeline"],
+   "detail-type": ["CodePipeline Pipeline Execution State Change", "CodePipeline Action Execution State Change", "CodePipeline Stage Execution State Change"],
+   "detail": {
+     "pipeline": ["first-pipeline", "second-pipeline"]
+   }
+ }
+ ```
+
+The event pattern above sets up the integration only for the `first-pipeline` and `second-pipeline` pipelines.
+
+### Correlate pipelines with tests
+
+If you are using the [Test Visibility][8] product and your pipeline contains one or more [AWS CodeBuild][9] actions to execute tests, you can correlate your tests
+with the related pipeline inside Datadog's CI Visibility product. To do that:
+1. In the AWS Console, go to your pipeline configuration and click on **Edit**
+2. Go to the stage containing the AWS CodeBuild action, click on **Edit Stage**, and then edit the relevant action.
+3. Under **Environment variables**, add a new environment variable. The name of the variable should be `DD_PIPELINE_EXECUTION_ID`
```

Comment:
"Name the variable `DD_PIPELINE_EXECUTION_ID`, and the value `#{codepipeline.PipelineExecutionId}`. Leave the type as _Plaintext_."